migrations/20210416143836-add-kind-to-transactions.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    // 1. Add the 'kind' column on Transactions table
+    await queryInterface.addColumn('Transactions', 'kind', {
+      allowNull: true,
+      type: Sequelize.ENUM(
+        'ADDED_FUNDS',
+        'CONTRIBUTION',
+        'EXPENSE',
+        'HOST_FEE',
+        'HOST_FEE_SHARE',
+        'PAYMENT_PROCESSOR_FEE',
+        'PLATFORM_FEE',
+        'PLATFORM_TIP',
+        'PREPAID_PAYMENT_METHOD',
+      ),
+    });
+
+    // 2. Pre-fill the 'kind' columns everywhere we can
+    // ADDED_FUNDS
+    await queryInterface.sequelize.query(`
+      UPDATE "Transactions" t
+      SET "kind" = 'ADDED_FUNDS'
+      FROM "PaymentMethods" pm
+      WHERE t."PaymentMethodId" = pm.id
+      AND pm.service = 'opencollective'
+      AND pm."type" = 'host'
+    `);
+
+    // CONTRIBUTION
+    // This will set KIND=CONTRIBUTION on all Transactions with an order, except for the ones
+    // we already migrated above (added funds) and prepaid budgets (tier #15445).
+    await queryInterface.sequelize.query(`
+      UPDATE "Transactions" t
+      SET "kind" = 'CONTRIBUTION'
+      FROM "Orders" o
+      WHERE t."OrderId" IS NOT NULL
+      AND t."OrderId" = o.id
+      AND o."TierId" != 15445
```

Comment:
I think it's ok if the initial payment for a pre-paid budget is a CONTRIBUTION.

What should be `PREPAID_PAYMENT_METHOD` is the transaction to give back and this one should not have have an order id?

Reply:
You're right. Actually, transactions created like that will have kin=CONTRIBUTION so we should not check that in the migration. `PREPAID_PAYMENT_METHOD` is properly set below and in `server/lib/prepaid-budget.js`